extensions/amp-analytics/0.1/visibility-manager.js

```diff
@@ -375,6 +411,18 @@ export class VisibilityManager {
           'elementHeight': layoutBox.height,
         }));
       }
+
+      state['initialScrollDepth'] = this.getRelativePosition_(
+          layoutBox.top,
+          layoutBox.height,
+          model.getInitialScrollDepth(),
+          viewport.getHeight());
+      state['maxScrollDepth'] = this.getRelativePosition_(
+          layoutBox.top,
+          layoutBox.height,
+          this.getMaxScrollDepth(),
+          viewport.getHeight());
+
```

Comment:
One concern I have here: we're doing math involving the initial and max scroll depths, which happened at some previous point in time; and the viewport height as of now.
So for example, let's say the top of the ad is at 1000px, the user scrolls down to position 800 when the viewport height was 100 (so 800...900 are visible, thus the ad is not visible), then scrolls up to position 300, and resizes the window to height 500. Now, this would say max scroll pos was 800, viewport height is 500, and therefore since 800<1000<(800+500) then max scroll depth is AT the ad.
The way to fix this would be to track 4 things instead of the current 2: instead of tracking initialScrollDepth and maxScrollDepth, we could track initialViewportTop, initialViewportBottom, maxViewportTop, and maxViewportBottom. Thoughts on this? Overkill?